packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -604,9 +604,14 @@ class RenderShaderMask extends RenderProxyBox {
     markNeedsPaint();
   }
 
+  bool get alwaysNeedsCompositing => child != null;
```

Comment:
you need to update the bits when child changes from null to non-null or vice-versa